### PR TITLE
[SPARK-6660][MLLIB] pythonToJava doesn't recognize object arrays

### PR DESCRIPTION
**Pull request by mengxr**
@davies

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/mllib/api/python/PythonMLLibAPI.scala [hunk 1]
```diff
@@ -1113,7 +1113,10 @@ private[spark] object SerDe extends Serializable {
       iter.flatMap { row =>
         val obj = unpickle.loads(row)
         if (batched) {
-          obj.asInstanceOf[JArrayList[_]].asScala
+          obj match {
+            case list: JArrayList[_] => list.asScala
+            case arr: Array[_] => arr
```
Comment (davies) [inline mllib/src/main/scala/org/apache/spark/mllib/api/python/PythonMLLibAPI.scala]: Does this match with `[]Byte` ?

I saw something like this
```
  case c if c.getClass.isArray => 
```

Not sure it's necessary or not.

Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/mllib/api/python/PythonMLLibAPI.scala]: Yes. This matches all Java Arrays.

**Comment by davies** [comment]
LGTM, thanks!

**Comment by mengxr** [comment]
Merged into master. The failed test is an unrelated yarn test.

**Pull request closed without merging**